contentcuration/contentcuration/frontend/channelList/views/KCard/BaseCard.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+
+  <li
+    :class="['remove-list-style card',$computedClass(coreOutlineFocus)]"
+    tabindex="0"
+    data-focus="true"
+    @focus="cardFocus"
+    @hover="cardHover"
+  >
+    <component
+      :is="headerLevel"
+    >
+      <KTextTruncator
+        v-if="title !== null"
+        :text="title"
+        :maxLines="titleLines"
+        class="spacing"
+      />
+    </component>
+    <slot name="default"></slot>
+  </li>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'BaseCard',
+    props: {
+      title: {
+        type: String,
+        required: false,
+        default: null,
+      },
+      headingLevel: {
```

Comment:
[Nice to have] We'd ideally validate here that the heading level is value between `2` and `6`.

Relatedly, I can see you added some basic validation in `KCard` and generally, for APIs that are the same in `BaseCard` and `KCard`, I'd recommend you move them to `BaseCard`. I'd say that logically it makes a bit more sense to have such validations in `BaseCard` since it is primarily this component that uses them to do something. The result for developers will be the same - in both cases they will see a warning in a console.